enterprise/com/src/main/java/org/neo4j/com/storecopy/TransactionCommittingResponseUnpacker.java

```diff
@@ -84,40 +99,87 @@ public void visit( CommittedTransactionRepresentation transaction, TxHandler han
         {
             long transactionId = transaction.getCommitEntry().getTxId();
             TransactionRepresentation representation = transaction.getTransactionRepresentation();
-            try
+            commitmentAccess.get().publishAsCommitted();
+            try ( LockGroup locks = new LockGroup();
+                  ValidatedIndexUpdates indexUpdates = indexUpdatesValidator.validate( representation ) )
             {
-                commitmentAccess.get().publishAsCommitted();
-                try ( LockGroup locks = new LockGroup();
-                      ValidatedIndexUpdates indexUpdates = indexUpdatesValidator.validate( representation, EXTERNAL ) )
-                {
-                    storeApplier.apply( representation, indexUpdates, locks, transactionId, EXTERNAL );
-                    handler.accept( transaction );
-                }
+                storeApplier.apply( representation, indexUpdates, locks, transactionId, EXTERNAL );
+                handler.accept( transaction );
             }
-            finally
+        }
+    };
+    private final TransactionVisitor batchCloser = new TransactionVisitor()
+    {
+        @Override
+        public void visit( CommittedTransactionRepresentation transaction, TxHandler handler,
+                Access<Commitment> commitmentAccess ) throws IOException
+        {
+            if ( commitmentAccess.get().markedAsCommitted() )
             {
-                transactionIdStore.transactionClosed( transactionId );
+                transactionIdStore.transactionClosed( transaction.getCommitEntry().getTxId() );
             }
         }
     };
     private TransactionAppender appender;
-    private TransactionRepresentationStoreApplier storeApplier;
+    private BatchingTransactionRepresentationStoreApplier storeApplier;
     private IndexUpdatesValidator indexUpdatesValidator;
     private TransactionIdStore transactionIdStore;
     private TransactionObligationFulfiller obligationFulfiller;
     private LogFile logFile;
     private LogRotation logRotation;
-    private volatile boolean stopped = false;
+    private volatile boolean stopped;
     private KernelHealth kernelHealth;
+    private final Function<DependencyResolver,BatchingTransactionRepresentationStoreApplier> applierFunction;
 
-    public TransactionCommittingResponseUnpacker( DependencyResolver resolver )
+    public TransactionCommittingResponseUnpacker( DependencyResolver resolver, int maxBatchSize )
     {
-        this( resolver, DEFAULT_BATCH_SIZE );
+        this.resolver = resolver;
+        this.validatorFunction = new Function<DependencyResolver,IndexUpdatesValidator>()
```

Comment:
Would it make code more readable if  instead of creating functions for constructing entities we will extract factories instead? wdyt?